### DISCUSSION TITLE
Access to external MongoDB using certificates

### DISCUSSION THREAD

**Discussion by daguadofernandez** [strapi/strapi#6177] [Legacy Topics]
Hi:

I´m trying to connect Strapi to a MongoDB cluster with certicates. My database.json is:

{
  "defaultConnection": "default",
  "connections": {
    "default": {
      "connector": "strapi-hook-mongoose",
      "settings": {
        "uri": "mongodb://myuser:mypassword@mongo-cluster-node01.domain.com:27017,mongo-cluster-node02.domain.com:27017,mongo-cluster-node03.domain.com:27017/ddbbname?replicaSet=replicasetname&ssl=true&sslCAFile=/tmp/certificates/mongoCA.crt&sslPEMKeyFile=/tmp/certificates/mycertificate.domain.com.pem&authSource=admin"
      },
      "options": {
        "ssl": false
      }
    }
  }
}

Is supported access a MongoDB with a URI and certificates? is correct this connection string?

Thanks

**Comment by derrickmehaffy** [comment]
`"connector": "strapi-hook-mongoose",`

You are using a fairly old version of Strapi if you still have the hook set there. What version are you running?

**Reply by daguadofernandez** [reply]
Hi...

Sorry by the delay... the version is strapi 3.0.0-alpha.26.2 

I have try with "mongoose" connector:

root@vmstrapi:~/myproject# strapi start
[2020-05-12T17:01:06.712Z] debug ⛔️ Server wasn't able to start properly.
[2020-05-12T17:01:06.712Z] error (hook:mongoose) takes too long to load
(hook:mongoose) takes too long to load


Regards.

**Reply by derrickmehaffy** [reply]
`strapi 3.0.0-alpha.26.2` This version is no longer supported and hasn't been in over a year.

**Reply by daguadofernandez** [reply]
Hi:

I have use latest version 3.0.0-beta.20.3

![image](https://user-images.githubusercontent.com/45290292/81805740-b154c600-951b-11ea-909b-a1d42a87ee43.png)




{
"defaultConnection": "default",
"connections": {
"default": {
"connector": "mongoose",
"settings": {
"uri": "mongodb://myuser:mypassword@mongo-cluster-node01.domain.com:27017,mongo-cluster-node02.domain.com:27017,mongo-cluster-node03.domain.com:27017/ddbbname?replicaSet=replicasetname&ssl=true&sslCAFile=/tmp/certificates/mongoCA.crt&sslPEMKeyFile=/tmp/certificates/mycertificate.domain.com.pem&authSource=admin"
},
"options": {
"ssl": false
}
}
}
}

If i use a MongoDb standalone very basic connect without problem.

![image](https://user-images.githubusercontent.com/45290292/81806090-4657bf00-951c-11ea-9207-eb477ff5e317.png)


Regards.

**Reply by sunnysonx** [reply]
Does the option "ssl": false conflict with SSL true from Mongo string?

**Reply by daguadofernandez** [reply]
Hi:

I have try with both options "true" and "false" and the result is the same:

![image](https://user-images.githubusercontent.com/45290292/81907634-5b8a2780-95c8-11ea-91ad-dcd9ff7b5f29.png)


Reagards.

**Reply by sunnysonx** [reply]
@daguadofernandez I had the same issue a few days ago, in my case the problem was in the password of Mongo, as I copied it Encoded from Mongo string URL, but Strapi accepts it only as Decoded.

**Reply by sunnysonx** [reply]
Also, can you connect to your external MongoDB from Mongo Compass? If you can, problem is 100% the password, but if you can't, then you should verify if your Mongo instance has external access for port 27017. Also, be aware to whitelist your server's IP and allow access only from it.

**Reply by daguadofernandez** [reply]
Hi:

I´m using decode password, in text plane.

I can connect from Mongo Compass and Studio 3D without problem.

**Reply by sunnysonx** [reply]
So that's the problem, try to pass the password as encoded in strapi.

**Reply by daguadofernandez** [reply]
Hi:

How encode?? base64???

Thanks

**Reply by sunnysonx** [reply]
Sorry, my bad, it should be decoded in strapi* and encoded in Mongo compass.
In strapi I have it stored like this: 
`"password": "1N@3ZbFE3@e"`
But in Mongo compass I use it like this:
`1N%403ZbFE3%40e`

**Reply by sunnysonx** [reply]
Also, make sure you use the right environment. Maybe you use development, but you set the password only for production.

**Comment by derrickmehaffy** [comment]
Locking as migrated to: https://forum.strapi.io/t/access-to-external-mongodb-using-certificates-6177/1863
